WordPress/Classes/Utility/Migration/ContentMigrationCoordinator.swift

```diff
@@ -0,0 +1,105 @@
+/// Encapsulates logic related to content migration from WordPress to Jetpack.
+///
+class ContentMigrationCoordinator {
+
+    static let shared: ContentMigrationCoordinator = .init()
+
+    // MARK: Dependencies
+
+    private let dataMigrator: ContentDataMigrating
+    private let keyValueDatabase: KeyValueDatabase
+    private let eligibilityProvider: ContentMigrationEligibilityProvider
+
+    init(dataMigrator: ContentDataMigrating = DataMigrator(),
+         keyValueDatabase: KeyValueDatabase = UserDefaults.standard,
+         eligibilityProvider: ContentMigrationEligibilityProvider = AppConfiguration()) {
+        self.dataMigrator = dataMigrator
+        self.keyValueDatabase = keyValueDatabase
+        self.eligibilityProvider = eligibilityProvider
+    }
+
+    enum ContentMigrationCoordinatorError: Error {
+        case ineligible
+        case exportFailure
+    }
+
+    // MARK: Methods
+
+    /// Starts the content migration process of exporting app data to the shared location
+    /// that will be accessible by the Jetpack app.
+    ///
+    /// The completion block is intentionally called regardless of whether the export process
+    /// succeeds or fails. Since the export process consists of local file operations, we should
+    /// just let the user continue with the original intent in case of failure.
+    ///
+    /// - Parameter completion: Closure called after the export process completes.
+    func startAndDo(completion: ((Result<Void, ContentMigrationCoordinatorError>) -> Void)? = nil) {
+        guard eligibilityProvider.isEligibleForMigration else {
+            completion?(.failure(.ineligible))
+            return
+        }
+
+        // TODO: Sync local post drafts here.
+
+        dataMigrator.exportData { result in
+            if case let .failure(error) = result {
+                DDLogError("[Jetpack Migration] Error exporting data: \(error)")
+                completion?(.failure(.exportFailure))
+                return
+            }
+
+            completion?(.success(()))
+        }
+    }
+
+    /// Silently starts the content migration process from WordPress to Jetpack.
+    /// This operation is only executed once per installation, and only performed
```

Comment:
I think we can remove the `This operation is only executed once per installation` as it is about the usage of the function. We can execute it as many times as we would want, only that it would return with empty completion.

Reply:
My point was more about the migration procedure — the fact that through this method, the `startAndDo` method will only be ever called once per installation, regardless of how often this method is called, is part of the scope of this method. 

But I agree the wording there is misleading. Good catch! 🙂